### ISSUE TITLE
Machine Learning Algorithms & Pipelines

### ISSUE THREAD

**Issue by KMouratidis** [KMouratidis/EDA_miner#2]
1. Add more algorithms:
- [ ] sklearn 
- [ ] [statsmodels](https://www.statsmodels.org/stable/index.html)
- [ ] [Edward](http://edwardlib.org/)
- [ ] tensorflow / pytorch
- [ ] [pyro](http://pyro.ai/)
- [ ] NLP models (nltk, gensim, spacy, etc)
- [ ] [PyMC](https://docs.pymc.io/)

2. Implement common pipelines, e.g.:
- [ ] Imputation -> Standardization -> Feature extraction -> Training
- [ ] Voting Classifiers, ensembles & other aggregators
- [ ] Cross-validation, (hyper-)parameter tuning
- [ ] Dimensionality reduction

**Comment by vikitsich** [comment]
@KMouratidis until we are ready with the model builder, I am going to redesign the layout of the analyze tabs. Ideally, we should output a graph (2D or 3D), a table with Accuracy, Specificity etc and a confusion matrix. Are you ok with that?

**Comment by KMouratidis** [comment]
Yes. Until we decide on how to integrate the model builder with the rest of the tabs (and if we will collapse them into one or not) feel free to experiment. Any updates won't go to waste; they might (at most) get moved elsewhere or abstracted.
